### PR TITLE
Limit setting

### PR DESCRIPTION
**Pull request by MartynasBas**
Added limit creation and modification logic to front-end and back-end.

### CODE REVIEW

**Comment by MartynasBas** [comment]
[Manager - Set limit logic](https://trello.com/c/m8rSQoVG/8-manager-set-limit-logic)

**Comment by MartynasBas** [comment]
[Manager - Set limit page](https://trello.com/c/fg2tyGy7/9-manager-set-limit-page)

File: SchedulearnBackend/SchedulearnBackend/Services/LimitService.cs [hunk 1]
```diff
@@ -1,8 +1,11 @@
-﻿using SchedulearnBackend.DataAccessLayer;
+﻿using Microsoft.EntityFrameworkCore;
+using SchedulearnBackend.Controllers.DTOs;
+using SchedulearnBackend.DataAccessLayer;
 using SchedulearnBackend.Models;
 using SchedulearnBackend.UserFriendlyExceptions;
 using System;
 using System.Collections.Generic;
+//using System.Data.Entity;
```
Comment (AinoraZ) [inline SchedulearnBackend/SchedulearnBackend/Services/LimitService.cs]: Remove

File: SchedulearnBackend/SchedulearnBackend/Migrations/20200527124346_UpdateLimit.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                nullable: true);
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropColumn(
+                name: "Name",
+                table: "Limits");
+
+            /*migrationBuilder.AddColumn<int>(
```
Comment (AinoraZ) [inline SchedulearnBackend/SchedulearnBackend/Migrations/20200527124346_UpdateLimit.cs]: Remove

File: schedulearn-frontend/src/components/LearningLimits/AssignLimits.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    const limitId = parseInt(event.target.value);
+    this.setState({ limitId });
+  }
+  render(): JSX.Element {
+    return (
+      <form onSubmit={(e): void => e.preventDefault()}>
+        <div>
+          <div>
+            <select id="limits" onChange={this.onLimitChange}>
+              <option value = {undefined} label = "Select limit" selected disabled={!this.props.undefinedLimitEnabled}/>
```
Comment (AinoraZ) [inline schedulearn-frontend/src/components/LearningLimits/AssignLimits.tsx]: No Limit

File: schedulearn-frontend/src/components/LearningLimits/AssignLimits.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+              {this.props.limits.map((limit): React.ReactNode => {
+                if(limit.name)
+                  return (<option value = {limit.id} 
+                    key = {limit.id} 
+                    label = {limit.name
```
Comment (AinoraZ) [inline schedulearn-frontend/src/components/LearningLimits/AssignLimits.tsx]: label={limit.name ? \`${limit.name} \`: "" 
\+ \`CD: ${limit.limitOfConsecutiveLearningDays} \`
\+ ...
}

File: schedulearn-frontend/src/components/LearningLimits/CreateLimits.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            />
+          </div>
+          <div>
+            <label>
+              Limit Of Learning Days Per Quarter 
+            </label>
+            <input type="text" 
+              placeholder="Per Quarter" 
+              onChange={this.onLimitOfLearningDaysPerQuarterChange} 
+              value={this.state.limitOfLearningDaysPerQuarter}
```
Comment (AinoraZ) [inline schedulearn-frontend/src/components/LearningLimits/CreateLimits.tsx:104]: make all inputs required

File: schedulearn-frontend/src/server-components/ManageLimitsViewer.tsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        body: JSON.stringify(limit),
+      })
+      .then(() => this.getLimits());
+  }
+
+
+  handleLimitUpdate = (limitId: number | undefined): void => {
+    if (!UserContext.user)
+      throw new Error("Should never reach this User Form when not logged in");
+    //console.log(this.state.currentUserId);
```
Comment (AinoraZ) [inline schedulearn-frontend/src/server-components/ManageLimitsViewer.tsx]: Remove

File: schedulearn-frontend/src/server-components/ManageLimitsViewer.tsx [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+  getLimits(): void {
+    if (!UserContext.user)
+      throw new Error("Should never reach Manage Limits view when not logged in");
+
+    UserContext
+      .fetch("api/Limit")
+      .then((data: Limit[]) => this.setState({ limits: data }));
+  }
+
+  getLimit(): void {
```
Comment (AinoraZ) [inline schedulearn-frontend/src/server-components/ManageLimitsViewer.tsx]: Unused, remove

**Review by AinoraZ — DISMISSED** [review]

**Review by AinoraZ — APPROVED** [review]

**Pull request merged**